src/sync.rs

```diff
@@ -0,0 +1,66 @@
+use std::collections::HashMap;
+
+pub type RoomId = String;
+
+#[derive(Deserialize, Clone)]
+pub struct SyncMsg {
+    #[serde(rename = "application/json")]
+    pub json: SyncMsgJson,
+}
+
+#[derive(Deserialize, Clone)]
+pub struct SyncMsgJson {
+    pub next_batch: String,
+    pub rooms: SyncRooms
+}
+
+#[derive(Deserialize, Clone)]
+pub struct SyncRooms {
+    pub invite: HashMap<RoomId, SyncRoomsInvite>,
+    pub join: HashMap<RoomId, SyncRoomsJoin>,
+}
+
+
+
+#[derive(Deserialize, Clone)]
+pub struct SyncRoomsInvite { }
+
+
+#[derive(Deserialize, Clone)]
+pub struct SyncRoomsJoin {
+    timeline: SyncRoomsJoinTimeline,
+}
+
+
+#[derive(Deserialize, Clone)]
+pub struct SyncRoomsJoinTimeline {
```

Comment:
Do we need all those long names, here? A matrix timeline should look the same everywhere in the matrix specs, doesn't it? So `Timeline` would suffice (or if not, `JoinTimeline` should).

I noted some of the types were already renamed in #4 … This should be done consistently.

(On a side note; why has the matrix API documentation become so horrible to read and search in? o_o I can't find the definition of a timeline object in there. Am I missing something?)